core/src/toga/window.py

```diff
@@ -300,16 +300,31 @@ def close(self) -> None:
         undefined, except for :attr:`closed` which can be used to check if the window
         was closed.
         """
+        close_window = True
         if self.app.main_window == self:
             # Closing the window marked as the main window is a request to exit.
             # Trigger on_exit handling, which may cause the window to close.
             self.app.on_exit()
-        else:
-            if self.content:
-                self.content.window = None
-            self.app.windows.discard(self)
-            self._impl.close()
-            self._closed = True
+            close_window = False
+        elif self.app.main_window is None:
+            # If this is a session-based app, this is the last window in the app,
+            # and the platform exits on last window close, trigger an exit.
+            if len(self.app.windows) == 1 and self.app._impl.CLOSE_ON_LAST_WINDOW:
```

Comment:
This should also happen if you set `main_window = None` but there are no open windows when `startup` returns.

Reply:
I had punted on this because when we get to document type handling, this is the trigger for "open a default document" behavior; but even then we're going to need *some* error handling, so I guess I can put in the initial stubs of error handling now.